query/stdlib/influxdata/influxdb/rules.go

```diff
@@ -764,3 +767,101 @@ func (PushDownWindowAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (p
 		WindowEvery:       window.Every.Nanoseconds(),
 	}), true, nil
 }
+
+//
+// Push Down of group  aggregates.
+// ReadGroupPhys |> { min, max, mean, count, sum }
+//
+type PushDownGroupAggregateRule struct{}
+
+func (PushDownGroupAggregateRule) Name() string {
+	return "PushDownWindowAggregateRule"
+}
+
+func (rule PushDownGroupAggregateRule) Pattern() plan.Pattern {
+	return plan.OneOf(
+		[]plan.ProcedureKind{
+			universe.MinKind,
+			universe.MaxKind,
+			universe.MeanKind,
+			universe.CountKind,
+			universe.SumKind,
+		},
+		plan.Pat(ReadGroupPhysKind))
+}
+
+func (PushDownGroupAggregateRule) Rewrite(ctx context.Context, pn plan.Node) (plan.Node, bool, error) {
+	// Check the aggregate function spec. Require operation on _value. There
+	// are two feature flags covering count and sum. All the other cases
+	// are specifically turned off.
+	var aggregateMethod string
+	fnNode := pn
+	switch fnNode.Kind() {
+	case universe.MinKind:
+		//	if !feature.PushDownSumAggregateRest().Enabled(ctx) || !caps.HaveMin() {
```

Comment:
I don't see this flag defined in `flags.yml`, perhaps you could add it in this PR?

Reply:
I'm a bit torn on this, some input from other team members would be good.

If you add the flag to flags.yml, but don't create the feature flag in config cat, I'm told the alerting system gets very noisy once the code is deployed. They said to make sure the flags are present in config cat before the code gets to production. I think it's something they want to fix.

So I wanted the code to be ready for the other aggregates to implement quickly when storage is ready, but I didn't want to fill up config config cat with feature flags covering features that are not currently being worked on. My rationale is that the flags are intended to be temporary so it would be good to not overpopulate it.

Reply:
The reason those cases are there is for testing the multi-function nature of the rewrite rules. That's so when we do want to add them we know it's an easy add. I agree that adding unused cases from the perspective of storage engine can lead to confusion and isn't so nice, however If we delete the unused cases we also have to delete the testing and that feels a bit like moving backwards, since the testing is already written and verifies that OneOf works as we expect.